src/core/styles/components/tiem-picker/_time-picker-theme.scss

```diff
@@ -0,0 +1,202 @@
+}
+
+
+/// @param {Map} $theme - The theme used to style the component.
+/// @param {Map} $palette [$default-palette] - The palette used as basis for styling the component.
+/// @requires igx-time-picker-theme
+/// @requires igx-color
+@mixin igx-time-picker($theme, $palette: $default-palette) {
+
+    $default-theme: igx-time-picker-theme(
+        $text-color: igx-color($palette, 'grays', 500),
+        $hover-text-color: igx-color($palette, 'secondary', 500),
+        $selected-text-color: igx-color($palette, 'secondary', 500),
+        $activeList-item-background: igx-color($palette, 'grays', 100),
+        $igx-time-picker-header-background: igx-color($palette, 'secondary', 500),
+        $igx-time-picker-header-hour-text-color: white,
+        $igx-time-picker-header-ampm-text-color: rgba(255, 255, 255, .8),
+        $background-color: white
+    );
+
+    $this: extend($default-theme, $theme);
+
+    %igx-time-picker__body {
+        display: flex;
+        padding: 10px 0;
+        justify-content: center;
+        background: map-get($this, 'background-color');
+    }
+
+    %time-picker-display {
+        max-width: 340px;
+        min-width: 320px;
+        padding: 0;
+    }
+
+    %time-picker-content {
+        width: 100%;
+        padding: 0;
+        color: inherit;
+        font-size: initial;
+        line-height: initial;
+    }
+
+    %time-picker-dialog-title {
+        display: none;
+    }
+
+    %time-picker-dialog-actions {
+        margin: 0;
+    }
+
+    %igx-time-picker__hourList {
+        text-align: right;
+    }
+
+    %igx-time-picker__minuteList {
+        text-align: center;
+    }
+
+    %igx-time-picker__ampmList {
+        display: flex;
+        flex-direction: column;
+        padding-top: 48px;
+    }
+
+    %igx-time-picker__column {
+        max-width: 64px;
+        height: 325px;
+        font-size: 16px;
+        padding: 0;
+        cursor: pointer;
+        display: flex;
+        flex-flow: column nowrap;
+        justify-content: space-between;
+        align-items: center;
+        color: map-get($this, 'text-color');
+        overflow: hidden;
+
+        &:focus,
+        &:active {
+            outline: none;
+        }
+    }
+
+    %igx-time-picker__item {
+        width: 54px;
+        padding: 5px 10px;
+        border-radius: 15px;
+        height: 48px;
+        display: flex;
+        justify-content: center;
+        align-items: center;
+
+        &:focus,
+        &:active {
+            outline: none;
+        }
+
+        &:hover {
+            color: map-get($this, 'hover-text-color');
+        }
+    }
+
+    %igx-time-picker__item--selected {
+        font-size: 24px;
+        color: map-get($this, 'selected-text-color');
+    }
+
+    %igx-time-picker__item--active {
+        background: map-get($this, 'activeList-item-background');
+    }
+
+    %igx-time-picker__header {
+        background: map-get($this, 'igx-time-picker-header-background');
+        padding: 1.5em 1em;
```

Comment:
Do we really want this to be set in relative values? And if so, why not use the em mixin?